subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputFilePropertiesIntegrationTest.groovy

```diff
@@ -28,11 +28,13 @@ class TaskInputFilePropertiesIntegrationTest extends AbstractIntegrationSpec {
             import org.gradle.api.internal.tasks.properties.GetInputFilesVisitor
             import org.gradle.api.internal.tasks.TaskPropertyUtils
             import org.gradle.api.internal.tasks.properties.PropertyWalker
+            import org.gradle.api.internal.file.FileResolver
 
             class CustomTask extends DefaultTask {
                 @Optional @$annotation.simpleName input
                 @TaskAction void doSomething() {
-                    GetInputFilesVisitor visitor = new GetInputFilesVisitor()
+                    def fileResolver = project.services.get(FileResolver)
+                    GetInputFilesVisitor visitor = new GetInputFilesVisitor(name, fileResolver)
```

Comment:
Where does `name` come from?

Reply:
From `Task.getName`. I could also choose an arbitrary name.

Reply:
Let's do that as it is a bit confusing without IDE support to figure out where it comes from, and I don't see why using `Task.name` would add anything in the first place.

Reply:
Arbitrary name it is.